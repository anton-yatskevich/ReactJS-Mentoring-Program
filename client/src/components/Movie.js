import React from 'react';
import PropTypes from 'prop-types';

const Movie = ({
    poster_path: path, title, release_date: date, vote_average: rating, tagline, runtime, overview
}) => (
    <div className="movie-content__wrapper">
        <div className="movie-content">
            <img className="movie-content__poster" src={path} alt={title} />
            <div className="movie-content__description">
                <h2 className="movie-content__title">{title}</h2>
                <span className="movie-content__rating">{rating}</span>
                <p className="movie-content__paragraph">{tagline}</p>
                <p className="movie-content__paragraph movie-content__paragraph--bold">
                    <span>{date.slice(0, 4)}</span>
                    <span>{`${runtime} min`}</span>
                </p>
                <p className="movie-content__paragraph">{overview}</p>
            </div>
        </div>
    </div>
);

Movie.propTypes = {
    poster_path: PropTypes.string,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    runtime: PropTypes.number,
    overview: PropTypes.string,
    tagline: PropTypes.string,
    release_date: PropTypes.string
};

Movie.defaultProps = {
    poster_path: '',
    title: '',
    tagline: '',
    runtime: 0,
    overview: '',
    vote_average: 0,
    release_date: ''
};

export default Movie;
