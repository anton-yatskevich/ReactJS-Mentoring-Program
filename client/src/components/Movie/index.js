import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Movie = ({
    poster_path: path, title, release_date: date, vote_average: rating, tagline, runtime, overview
}) => (
    <Wrapper>
        <Content>
            <Poster src={path} alt={title} />
            <Description>
                <Heading>{title}</Heading>
                <Rating>{rating}</Rating>
                <Paragraph>{tagline}</Paragraph>
                <ParagraphBold>
                    <ReleaseDate>{date.slice(0, 4)}</ReleaseDate>
                    <span>{`${runtime} min`}</span>
                </ParagraphBold>
                <Paragraph>{overview}</Paragraph>
            </Description>
        </Content>
    </Wrapper>
);

Movie.propTypes = {
    poster_path: PropTypes.string,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    runtime: PropTypes.number,
    overview: PropTypes.string,
    tagline: PropTypes.string,
    release_date: PropTypes.string
};

Movie.defaultProps = {
    poster_path: '',
    title: '',
    tagline: '',
    runtime: 0,
    overview: '',
    vote_average: 0,
    release_date: ''
};

const Content = styled.div`
    margin: 0 auto;
    max-width: 1100px;
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
`;

const Wrapper = styled.div`
    width: 100%;
    background-color: rgb(146, 133, 133);
    padding: 30px 20px;
`;

const Poster = styled.img`
    width: 320px;
    margin-right: 30px;
`;

const Description = styled.div`
    margin-top: 10px;
`;

const Heading = styled.h2`
    color: rgb(160, 0, 0);
    margin: 0 20px 15px 0;
    display: inline-block;
    font-size: 28px;
`;

const Rating = styled.span`
    border: 1px solid rgb(0, 100, 17);
    color: rgb(0, 100, 17);
    padding: 10px 8px;
    border-radius: 50%;
    font-size: 18px;
`;

const Paragraph = styled.p`
    font-size: 18px;
    margin-bottom: 20px;
`;

const ParagraphBold = styled(Paragraph)`
    font-weight: bold;
`;

const ReleaseDate = styled.span`
    margin-right: 30px;
`;

export default Movie;
