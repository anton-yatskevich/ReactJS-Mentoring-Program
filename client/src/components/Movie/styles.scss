.movie-content {
	margin: 0 auto;
	max-width: 1100px;
	display: flex;
	justify-content: flex-start;
	align-items: flex-start;

	&__wrapper {
		width: 100%;
		background-color: rgb(146, 133, 133);
		padding: 30px 20px;
	}

	&__poster {
		width: 320px;
		margin-right: 30px;
	}

	&__description {
		margin-top: 10px;
	}

	&__title {
		color: rgb(160, 0, 0);
		margin: 0 20px 15px 0;
		display: inline-block;
		font-size: 28px;

	}

	&__rating {
		border: 1px solid rgb(0, 100, 17);
		color: rgb(0, 100, 17);
		padding: 10px 8px;
		border-radius: 50%;
		font-size: 18px;
	}

	&__paragraph {
		font-size: 18px;
		margin-bottom: 20px;
		
		&--bold {
			font-weight: bold;
		}

		span:first-of-type {
			margin-right: 30px;
		}
	}
}
