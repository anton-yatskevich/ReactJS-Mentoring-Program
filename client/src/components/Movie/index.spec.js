import React from 'react';
import { shallow } from 'enzyme';
import Movie from './index';

describe('Movie component', () => {
    it('should be rendered with default props', () => {
        const MovieComponent = shallow(<Movie />);
        expect(MovieComponent).toMatchSnapshot();
    });
});
