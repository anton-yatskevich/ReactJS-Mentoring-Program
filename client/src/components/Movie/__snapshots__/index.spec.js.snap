// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should be rendered with default props 1`] = `
<div
  className="movie-content__wrapper"
>
  <div
    className="movie-content"
  >
    <img
      alt=""
      className="movie-content__poster"
      src=""
    />
    <div
      className="movie-content__description"
    >
      <h2
        className="movie-content__title"
      />
      <span
        className="movie-content__rating"
      >
        0
      </span>
      <p
        className="movie-content__paragraph"
      />
      <p
        className="movie-content__paragraph movie-content__paragraph--bold"
      >
        <span />
        <span>
          0 min
        </span>
      </p>
      <p
        className="movie-content__paragraph"
      />
    </div>
  </div>
</div>
`;
